### PR TITLE
Unify common grid create/update POST/PUT parameters

### PR DESCRIPTION
**Pull request by SpComb**
Update the `kontena grid create` CLI, `POST /v1/grids` API and server `Grids::Create` mutations to accept the same options as previously accepted by `kontena grid update`, `PUT /v1/grids/:grid` and `Grids::Update`:

* `--default-affinity` -> `{ "default_affinity": [ ... ] }` (this was already supported)
* `{ "trusted_subnets": [ ... ] }` (not exposed in `kontena grid create`, use `kontena grid trusted-subnet`)
* `--statsd-server` -> `{ "stats": { "statsd": ... } }`
* `--log-forwarder` `--log-opts` -> `{ "logs": { ... } }`

#### `kontena grid create`
```
Usage:
    kontena grid create [OPTIONS] NAME

Parameters:
    NAME                          Grid name

Options:
    --initial-size INITIAL_SIZE   Initial grid size (number of nodes) (default: 1)
    --silent                      Reduce output verbosity
    --token [TOKEN]               Set grid token
    --subnet [CIDR]               Configure grid overlay subnet
    --supernet [CIDR]             Configure grid IPAM supernet
    --default-affinity [AFFINITY] Default affinity rule for the grid
    --statsd-server STATSD_SERVER Statsd server address (host:port)
    --log-forwarder LOG_FORWARDER Set grid wide log forwarder (set to 'none' to disable)
    --log-opt [LOG_OPT]           Set log options (key=value)
    -h, --help                    print help
    -D, --debug                   Enable debug (default: $DEBUG)
```

#### `kontena grid update`
```
Usage:
    kontena grid update [OPTIONS] NAME

Parameters:
    NAME                          Grid name

Options:
    --default-affinity [AFFINITY] Default affinity rule for the grid
    --statsd-server STATSD_SERVER Statsd server address (host:port)
    --log-forwarder LOG_FORWARDER Set grid wide log forwarder (set to 'none' to disable)
    --log-opt [LOG_OPT]           Set log options (key=value)
    --no-default-affinity         Unset grid default affinity
    --no-statsd-server            Unset statsd server setting
    -h, --help                    print help
    -D, --debug                   Enable debug (default: $DEBUG)
```

### CODE REVIEW

**Comment by SpComb** [comment]
Testing the new CLI options:

```
$ bundle exec bin/kontena grid create --subnet 10.79.0.0/16 --default-affinity label==test --statsd-server 127.0.0.1 --log-forwarder fluentd --log-opt fluentd-address=127.0.0.1 test-options
 [warn] Option --initial-size=1 is only recommended for test/dev usage
 [done] Creating test-options grid      
 [done] Switching scope to test-options grid      
$ bundle exec bin/kontena grid show test-options
test-options:
  uri: ws://localhost:9292//
  initial_size: 1
  default_affinity: 
    - label==test
  subnet: 10.79.0.0/16
  supernet: 10.80.0.0/12
  stats:
    nodes: 0 of 0
    cpus: 0
    load: 0.0 0.0 0.0
    memory: 0.0 of 0.0 GB
    filesystem: 0.0 of 0.0 GB
    users: 1
    services: 0
    containers: 0
  exports:
    statsd: 127.0.0.1:8125
logs:
  forwarder: fluentd
  fluentd-address: 127.0.0.1
```

File: cli/lib/kontena/cli/grids/common.rb [hunk 1]
```diff
@@ -117,5 +117,68 @@ def get_grid(name = nil)
         exit_with_error "No grid given or selected"
       end
     end
+
+    module Parameters
+      def self.included(base)
+        base.class_eval do
```
Comment (kke) [inline cli/lib/kontena/cli/grids/common.rb]: you can also just do:

```ruby
def self.included(base)
  base.option "foofoo", :flag, "bar"
end
```

File: cli/lib/kontena/cli/grids/common.rb [hunk 2]
```diff
@@ -117,5 +117,68 @@ def get_grid(name = nil)
...
+        end
+      end
+
+      def validate_log_opts
+        if !log_opt_list.empty? && log_forwarder.nil?
+          raise Kontena::Errors::StandardError.new(1, "Need to specify --log-forwarder when using --log-opt")
+        end
+      end
+
+      def parse_log_opts
```
Comment (kke) [inline cli/lib/kontena/cli/grids/common.rb]: option normalizing can be done in a block:

```ruby
option '--log-opt', foofoo do |log_opt|
  k,v = log_opt.split('=')
end
```

File: cli/lib/kontena/cli/grids/create_command.rb [hunk 3]
```diff
@@ -10,26 +10,31 @@ class CreateCommand < Kontena::Command
...
       payload[:token] = self.token if self.token
       payload[:initial_size] = self.initial_size if self.initial_size
-      payload[:default_affinity] = self.default_affinity_list unless self.default_affinity_list.empty?
       payload[:subnet] = subnet if subnet
       payload[:supernet] = supernet if supernet
-      
+
+      build_grid_parameters(payload)
+
       grid = nil
```
Comment (kke) [inline cli/lib/kontena/cli/grids/create_command.rb]: not changed in this PR, but the spinner returns the outcome:
```ruby
grid = spinner "foofoo" {  client.post('grids',  payload) }
```

File: cli/lib/kontena/cli/grids/common.rb (hunk 2 above)
Comment (SpComb) [inline cli/lib/kontena/cli/grids/common.rb]: Not exactly the same thing though, because this also collects those into a single `Hash`?

Comment (kke) [inline cli/lib/kontena/cli/grids/common.rb]: yep, not sure how it works with multivalued options.

File: cli/lib/kontena/cli/grids/create_command.rb [hunk 4]
```diff
@@ -10,28 +10,32 @@ class CreateCommand < Kontena::Command
...
     option "--supernet", "[CIDR]", "Configure grid IPAM supernet"
 
+    include Common::Parameters
+
     requires_current_master_token
 
     def execute
+      validate_grid_parameters
+
+      if initial_size == 1
```
Comment (jakolehm) [inline cli/lib/kontena/cli/grids/create_command.rb]: This reminds me that we should prompt initial size from user (separate pr).

**Pull request merged**